src/TraceEvent/ETWTraceEventSource.cs

```diff
@@ -55,6 +56,21 @@ public ETWTraceEventSource(string fileOrSessionName, TraceEventSourceType type)
             Initialize(fileOrSessionName, type);
         }
 
+        /// <summary>
+        /// Open multiple etl files as one trace for processing.
+        /// </summary>
+        /// <param name="fileOrSessionName">
+        /// If type == MergeAll, call Initialize
+        /// <param name="type"></param>
+        // [SecuritySafeCritical]
+        public ETWTraceEventSource(IEnumerable<string> fileNames, TraceEventSourceType type)
+        {
```

Comment:
Before checking the type, you should unconditionally set `this.fileNames = fileNames` like the other constructor does, since if `type != TraceEventSourceType.MergeAll` then the file names won't get saved.  Basically if `type == TraceEventSourceType.FileOnly` then we want to replicate the behavior of `ETWTraceEventSource..ctor(IEnumerable<string> fileNames)`.

Reply:
Adding to the else clause when type != MergeAll